lib/ably/Presence.php

```diff
@@ -0,0 +1,50 @@
+<?php
+require_once 'PaginatedResource.php';
+
+class Presence {
+
+    private $name;
+    private $channelPath;
+    private $ably;
+
+    /**
+     * Constructor
+     * @param AblyRest $ably Ably API instance
+     * @param string $name Channel's name
+     */
+    public function __construct( AblyRest $ably, $name ) {
+        $this->ably = $ably;
+        $this->name = urlencode($name);
```

Comment:
Shouldn't you be storing the unencoded name and encoding in the channelPath?  This way round there is no way to know what the actual channel name is.